test/use-semicolon-delimiter.test.js

```diff
@@ -1,110 +1,101 @@
 'use strict'
 
-const t = require('tap')
-const test = t.test
+const { test } = require('node:test')
 const Fastify = require('..')
 const sget = require('simple-get').concat
 
 test('use semicolon delimiter default false', t => {
-  t.plan(4)
-
   const fastify = Fastify({})
 
-  t.teardown(fastify.close.bind(fastify))
+  t.after(fastify.close())
 
   fastify.get('/1234;foo=bar', (req, reply) => {
     reply.send(req.query)
   })
 
   fastify.listen({ port: 0 }, err => {
-    t.error(err)
+    t.assert.ifError(err)
 
     sget({
       method: 'GET',
       url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
     }, (err, response, body) => {
-      t.error(err)
-      t.equal(response.statusCode, 200)
-      t.same(JSON.parse(body), {})
+      t.assert.ifError(err)
+      t.assert.strictEqual(response.statusCode, 200)
+      t.assert.deepStrictEqual(JSON.parse(body), {})
     })
   })
 })
 
 test('use semicolon delimiter set to true', t => {
-  t.plan(4)
-
   const fastify = Fastify({
     useSemicolonDelimiter: true
   })
-  t.teardown(fastify.close.bind(fastify))
+  t.after(fastify.close())
 
   fastify.get('/1234', (req, reply) => {
     reply.send(req.query)
   })
 
   fastify.listen({ port: 0 }, err => {
-    t.error(err)
+    t.assert.ifError(err)
 
     sget({
       method: 'GET',
       url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
     }, (err, response, body) => {
-      t.error(err)
-      t.equal(response.statusCode, 200)
-      t.same(JSON.parse(body), {
+      t.assert.ifError(err)
+      t.assert.strictEqual(response.statusCode, 200)
+      t.assert.deepStrictEqual(JSON.parse(body), {
         foo: 'bar'
       })
     })
   })
 })
 
 test('use semicolon delimiter set to false', t => {
-  t.plan(4)
-
   const fastify = Fastify({
     useSemicolonDelimiter: false
   })
-  t.teardown(fastify.close.bind(fastify))
+  t.after(fastify.close())
 
   fastify.get('/1234;foo=bar', (req, reply) => {
     reply.send(req.query)
   })
 
   fastify.listen({ port: 0 }, err => {
-    t.error(err)
+    t.assert.ifError(err)
 
     sget({
       method: 'GET',
       url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
     }, (err, response, body) => {
-      t.error(err)
-      t.equal(response.statusCode, 200)
-      t.same(JSON.parse(body), {})
+      t.assert.ifError(err)
+      t.assert.strictEqual(response.statusCode, 200)
+      t.assert.deepStrictEqual(JSON.parse(body), {})
     })
   })
 })
 
 test('use semicolon delimiter set to false return 404', t => {
```

Comment:
Call `done` in after last test assertion
```suggestion
test('use semicolon delimiter set to false return 404', (t, done) => {
  t.plan(3)
```

Reply:
Adding the `done()` returns me this:

```
Error [ERR_TEST_FAILURE]: The resolution of the promise is still pending but the event loop has already been resolved
at process.emit (node:events:519:28) {
code: ‘ERR_TEST_FAILURE’,
failureType: ‘cancelledByParent’,
cause: ‘The resolution of the promise is still in progress, but the event loop has already been resolved’.
}
 ```

Reply:
Interesting not sure why that is not working....

This works calling `fastify.close()` before `done()`
```js
'use strict'

const { test } = require('node:test')
const Fastify = require('..')
const sget = require('simple-get').concat

test('use semicolon delimiter default false', (t, done) => {
  t.plan(4)

  const fastify = Fastify()

  fastify.get('/1234;foo=bar', (req, reply) => {
    console.log('helloooo')
    reply.send(req.query)
  })

  fastify.listen({ port: 0,  }, err => {
    t.assert.ifError(err)
    sget({
      method: 'GET',
      url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
    }, (err, response, body) => {
        console.error(err)
      t.assert.ifError(err)
      t.assert.strictEqual(response.statusCode, 200)
      t.assert.deepStrictEqual(JSON.parse(body), {})
      fastify.close()
      done()
    })
  })
})

test('use semicolon delimiter set to true', (t, done) => {
  t.plan(4)
  const fastify = Fastify({
    useSemicolonDelimiter: true
  })

  fastify.get('/1234', (req, reply) => {
    reply.send(req.query)
  })

  fastify.listen({ port: 0 }, err => {
    t.assert.ifError(err)

    sget({
      method: 'GET',
      url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
    }, (err, response, body) => {
      t.assert.ifError(err)
      t.assert.strictEqual(response.statusCode, 200)
      t.assert.deepStrictEqual(JSON.parse(body), {
        foo: 'bar'
      })
      fastify.close()
      done()
    })
  })
})

test('use semicolon delimiter set to false', (t, done) => {
  t.plan(4)

  const fastify = Fastify({
    useSemicolonDelimiter: false
  })

  fastify.get('/1234;foo=bar', (req, reply) => {
    reply.send(req.query)
  })

  fastify.listen({ port: 0 }, err => {
    t.assert.ifError(err)

    sget({
      method: 'GET',
      url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
    }, (err, response, body) => {
      t.assert.ifError(err)
      t.assert.strictEqual(response.statusCode, 200)
      t.assert.deepStrictEqual(JSON.parse(body), {})
      fastify.close()
      done()
    })
  })
})

test('use semicolon delimiter set to false return 404', (t, done) => {
  t.plan(3)

  const fastify = Fastify({
    useSemicolonDelimiter: false
  })

  fastify.get('/1234', (req, reply) => {
    reply.send(req.query)
  })

  fastify.listen({ port: 0 }, err => {
    t.assert.ifError(err)

    sget({
      method: 'GET',
      url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
    }, (err, response, body) => {
      t.assert.ifError(err)
      t.assert.strictEqual(response.statusCode, 404)
      fastify.close()
      done()
    })
  })
})
```

Reply:
very interesting

Reply:
ok, now the test passes all the checks, we will see if this solution will be the final one.

Thanks for the help 🙂